src/libraries/Microsoft.PowerFx.Core/Public/DisplayNames/DisplayNameProvider.cs

```diff
@@ -15,5 +15,14 @@ internal abstract class DisplayNameProvider
         public abstract bool TryGetLogicalName(DName displayName, out DName logicalName);
 
         public abstract bool TryGetDisplayName(DName logicalName, out DName displayName);
+
+        /// <summary>
+        /// This function attempts to remap logical and display names given a display name.
+        /// It's used for scenarios where display names are changed under the hood while the expression is in display name format already.
+        /// This is a legacy Canvas app behavior, and should not be supported implemented by non-canvas hosts.
+        /// If this isn't supported by a given display name provider, this should return the same as 
+        /// <see cref="TryGetLogicalName(DName, out DName)"/>, with the newDisplayName output populated by the first arg. 
+        /// </summary>
+        public abstract bool TryRemapLogicalAndDisplayNames(DName displayName, out DName logicalName, out DName newDisplayName);
```

Comment:
>TryRemapLogicalAndDisplayNames [](http://example.com/codeflow?start=29&length=30)

Why not have this call TryGetLogicalName by default? Can this class be non-abstract? Canvas provider can overwrite it.